src/extension.ts

```diff
@@ -100,18 +100,25 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
             await ext.tree.refresh(ext.attachedAccountsNode);
         });
         registerCommand('azureDatabases.refresh', async (_actionContext: IActionContext, node?: AzExtTreeItem) => await ext.tree.refresh(node));
-        registerCommand('cosmosDB.detachDatabaseAccount', async (actionContext: IActionContext, node?: AzureTreeItem) => {
-            if (!node) {
-                node = await ext.tree.showTreeItemPicker<AzureTreeItem>(cosmosDBTopLevelContextValues.map((val: string) => val += AttachedAccountSuffix), actionContext);
-            }
-            if (node instanceof MongoAccountTreeItem) {
-                if (ext.connectedMongoDB && node.fullId === ext.connectedMongoDB.parent.fullId) {
-                    setConnectedNode(undefined, codeLensProvider);
-                    await node.refresh();
+        registerCommand('cosmosDB.detachDatabaseAccount', async (actionContext: IActionContext & ITreeItemPickerContext, node?: AzureTreeItem) => {
+            const numChildren: number = ext.attachedAccountsNode.loadAllChildren.length;
+            if (numChildren < 2) {
```

Comment:
This logic works if the only child is the "Attach Database Account..." node. But if you have a single attached account you won't be able to detach it.

Maybe instead you could check the context value of the child node here before showing that info message?